File: FirstPage/app/src/main/java/com/example/firstpage/MainActivity.java
package com.example.firstpage;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView first = findViewById(R.id.first_page);
        String text = (String) first.getText();

        first.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(getBaseContext(), SecondPage.class);
                i.putExtra("Hello_World", "Hello World..!!");
                startActivity(i);
            }
        });
    }
}

File: RupeekChallenge/app/src/main/java/com/example/rupeekchallenge/MainActivity.kt
package com.example.rupeekchallenge

import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.os.Handler

import android.view.WindowInsets
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {

    private val SPLASH_SCREEN_TIME_OUT = 2000

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        @Suppress("DEPRECATION")
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            window.insetsController?.hide(WindowInsets.Type.statusBars())
        } else {
            window.setFlags(
                    WindowManager.LayoutParams.FLAG_FULLSCREEN,
                    WindowManager.LayoutParams.FLAG_FULLSCREEN
            )
        }

        setContentView(R.layout.activity_main)

        @Suppress("DEPRECATION")
        Handler().postDelayed({
            val i = Intent(this@MainActivity,
                    HomeActivity::class.java)
            //Intent is used to switch from one activity to another.
            startActivity(i)
            //invoke the SecondActivity.
            finish()
            //the current activity will get finished.
        }, SPLASH_SCREEN_TIME_OUT.toLong())
    }
}